client/request.go

```diff
@@ -129,6 +130,12 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
 			return serverResp, err
 		}
 
+		if err, ok := err.(*net.OpError); ok {
+			if os.IsPermission(err.Err) {
+				return ServerResp, ErrPermissionDenied
```

Comment:
Would be nice to wrap the error via `errors.Wrap` with some context like the socket path.